search/includes/classes/class-search.php

```diff
@@ -1164,9 +1162,10 @@ public function ep_handle_failed_request( $request, $response, $query, $type, $c
 			$response_failure_code = $response->get_error_code();
 
 			foreach ( $error_messages as $error_message ) {
-				$reason = $this->is_curl_timeout( $error_message ) ? Prometheus_Collector::QUERY_FAILED_TIMEOUT : Prometheus_Collector::QUERY_FAILED_ERROR;
-
-				Prometheus_Collector::increment_failed_query_counter( $query_method, $query['url'], $reason );
+				if ( class_exists( Prometheus_Collector::class ) ) {
```

Comment:
I would move this check out of `foreach` loop: if the class does not exist, we don't need the loop.